clcache.py

```diff
@@ -232,6 +232,9 @@ def getManifestHash(compilerBinary, commandLine, sourceFile):
         # One of the few exceptions to this rule is the /MP switch, which only
         # defines how many compiler processes are running simultaneusly.
         commandLine = [arg for arg in commandLine if not arg.startswith("/MP")]
+        baseDir = os.environ.get('CLCACHE_BASEDIR')
+        if baseDir:
+            commandLine = [arg.replace(baseDir, BASEDIR_REPLACEMENT) for arg in commandLine]
```

Comment:
Indeed, the first three sound like good candidates. I suggest the replacement should be applied to their argument -- we can extend (or shorten) that list later as we learn about new cases.

`/Yu` doesn't seem important right now since clcache doesn't support invocations making use of precompiled headers anyway.